### ISSUE TITLE
Radio transmitters are straight fucked

### ISSUE THREAD

**Issue by Hoshi-chan** [d3athrow/vgstation13#6057]
During a round where my borgs and I had an ion law that stated that the station must have music, we spent two and a half hours attempting to fanagle the radio transmitters into a working state. Here are the variety of problems that we encountered:

* The radio transmitter produces OUTRAGEOUS amounts of heat, on top of the radiation that it's always produced. It required two separate sets of thermal plates on a space tile to cool the area enough for the radio transmitter to consistently function without immediately shutting down.

* The radio transmitter does not bounce a juke's signal, it hijacks it. This means that once a juke has been multitooled to a transmitter, that juke no longer produces ambient audio for the room that it's in, unless that room also has a sound system set to the appropriate frequency.

* The transmitter's interface has a frustrating tendency to break, completely. It can no longer be interacted with, have its frequency changed, its power toggled, or anything. This is difficult to reproduce. Just attempt to set a transmitter up; unwrench it, wrench it, fiddle with it, and find out.

* Once you screwdriver open the maintenance panel on a transmitter, it can no longer be closed, and you cannot use a multitool to bring the wire window up, only wirecutters, although they can then be pulsed with a multitool once you've used wirecutters to open the menu.

* Even with all of these problems, it sometimes fails to properly transmit to sound systems around the station, which is what we had the most trouble with.

**Comment by mrpain666** [comment]
Those things have always been an antag's wet dream for some reason.

**Comment by dylanstrategie** [comment]
Might have to do with the fact you can absolutely bake any crewman that comes within five meters of it without a full radiation suit, internals and a quick way out

Trust me, it was way worse when you could drag those around without turning it off

**Comment by Hoshi-chan** [comment]
I feel like both of you are missing the point, and I don't see how it's useful for antags. It doesn't put off that much radiation, and it only puts off heat very briefly before it becomes too hot and shuts down.

The problem is that it's buggy and doesn't work properly, the maintenance panel is broken, the interface breaks, it more or less nullifies a jukebox in order to work at all, and the cooling requirements are absurd, given that the one on the DJ Station begins in pristine condition, good to go, and seems to run off of pure bullshit, since it also lacks any of the sky-high requirements to get one to function anywhere else; it just works.

**Comment by Kurfursten** [comment]
- Unable to reproduce. Having no trouble keeping the radio transmitter cooled with a simple AC. Are you sure your pipes weren't borked?
- That's a feature, I assumed, though feel free to feature request it away. If you want music in the same room as a juke, build a sound system or buy a boombox.
- I suspect the cause of the third is that you've cut the LINK wire, which fucks with its multitool menu by making it not pop up at all. Please test further, I was able to get the unresponsiveness and fix it by mending the LINK wire.
- I was able to reproduce this, I'll make a fix.
- Are you sure that you set the output frequency to 101.5, which is what almost all sound systems on the station are set to?

**Comment by Hoshi-chan** [comment]
All three of us watched the radio transmitter heat the core to 400 degrees in under a minute and then shut down. It did the same thing to tcomms, which overheated and destroyed all of the machines. The MoMMI even commented, "What does this thing run on, shards of hell?"

In a previous round, with the same MoMMI, with me as an engiborg, we had set it up in the room that it spawns in with a freezer running to a passive vent. It never got cool enough to run for longer than a few seconds. We also experienced an issue where once it had overheated, it never cooled down enough to run again. It basically just stayed overheated forever. This happened intermittently after multiple failures, and was one of the reasons that we had to request admin intervention for new transmitters.

Also, you can BUILD sound systems? How? Now that I understand the way that it jacks the signal, I guess I'm fine with it, but I wish that a lot of the radio transmitter's functionality were more transparent, like maybe a note spawning in the room with it (and on the DJ station) with instructions on its use, its limitations, and the requirements for running it.

And yes, we tried multiple frequencies, and changing sound systems around the station to them. We were experiencing issues with 101.5 because it seemed to be conflicting with another transmitter (on the DJ station), which, once disabled, affected nothing. Then again, said transmitter's menu broke, and it was "running on pure bullshit" according to my borg, with no power, cooling, or anything else. It may have had to do with my borg's streaming settings, but even the MoMMI was saying that he was hearing different music in different places from what was ostensibly the same juke and transmitter.

Is there some way that we can throw up a server together to experiment with it, to see if I can't reproduce the heating issues? I feel like maybe I just have a gift for breaking shit.

**Comment by Kurfursten** [comment]
@Hoshi-chan Wait for #6084 to hit and we'll make an appointment for the N3X test server

**Comment by Faptastic** [comment]
I partly replicated our little ai core fun time on a local server and aside from the fact I was entirely lacking a music library for the jukebox to detect, managed to get a superjuke synched into the transmitter in it's little room working no problem and apparently spamming the hallways with it's non-funk as indicated by the seemingly endless barrage of test logs.
Heat was also controlled with a freezer no problem. I had the juke outside the transmitter room near a window.

This does not reflect the temperatures we endured during the round however or the absolutely insane behaviour of the transmitters interface which I very much doubt was related to a cut wire, we couldn't even unbolt the damn things after a while or use a screwdriver on em. The behavior of the transmitter was abnormal to say the least having interacted with it before.

The rooms were absolutely freezing, it was cold enough in the AI core to let it run nonstop.
It took down telecomms almost immediately when installed into the room, though it did actually work in there, at least I think it did.

I did also notice the backlight indicator is backwards, it's dim when powered when it should be the other way according to the wiki?

I would like to retry it on the main server during a round, all I need is a superjuke and the implied admin consent that comes with it.
